pytorch_lightning/plugins/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Callable, Dict, Mapping, Optional, TypeVar, Union
+
+TSaveStorageOptions = TypeVar("TSaveStorageOptions", Mapping, Callable)
+TLoadStorageOptions = TypeVar("TLoadStorageOptions", Mapping, Callable)
```

Comment:
MyPy still complaining about typing here and I'm clueless:

https://github.com/PyTorchLightning/pytorch-lightning/pull/8743/checks?check_run_id=3313010783#step:5:7

```
pytorch_lightning/plugins/checkpoint/torch.py:24: error: Argument 2 of "load_checkpoint" is incompatible with supertype "CheckpointIOPlugin"; supertype defines the argument type as "Optional[TLoadStorageOptions]"  [override]
pytorch_lightning/plugins/checkpoint/torch.py:24: note: This violates the Liskov substitution principle
pytorch_lightning/plugins/checkpoint/torch.py:24: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

Any ideas @Borda @carmocca @ananthsub @awaelchli?

Reply:
i see, the constructor of the plugin would  for the plugin would need to communicate this with Generics

```
class CheckpointIOPlugin(ABC, Generic[TSaveStorageOptions, TLoadStorageOptions]):
```

then the subclasses would be like:

```
TorchCheckpointIOPlugin(CheckpointIOPlugin[None, Callable]):
```

however, if you think this is too much boilerplate, we can rethink this or just allow any, and subclasses can add mypy overrides

Reply:
I've opted just for Any out of simplicity :)